### ISSUE TITLE
Add option to completely disable removal of newlines

### ISSUE THREAD

**Issue by andrewvaughan** [prettier/prettier#15524]
I believe versions of this have been thrown around from looking through the issues (although, over 1K open is hard to parse...). However, I think this situation is pretty cut and dry.

I enjoy everything about the opinionated type of approach Prettier takes, but finds it falls woefully short in appropriately choosing when to (and when to not) collapse newlines. I often get output like this that really hurts readability and formatting after running Prettier:

```
##
# Disables specific checks.
#
skip-check:
  ##
  # Ensure that containers have `HEALTHCHECK` instructions, the default from MegaLinter.
  #
  - CKV_DOCKER_2
```

It's not the end of the world, but having the comment blocks squashed like this means I have to get ...creative... to keep the file easily readable.

It also seems that Prettier is extremely inconsistent about when blank lines are collapsed and when they aren't. Some files have no issue having double newlines, some files collapse them right next to the code completely squashing and killing the readability of the code.

Whereas I believe most requests and recommendations have been about changing how Prettier's opinion is - my recommendation is more simple. Simply add a configuration option that completely disables newline formatting. This way I can maintain readability in my files as I wish, and as my comment needs differ between blocks, inline, etc. Keep the current opinions the default, but allow for that one piece of logic to be disabled.

Thoughts? Any other way to solve this that you can think of?

Thanks!

**Comment by sosukesuzuki** [comment]
We won't add any option for this. If Prettier's behavior is clearly incorrect, please report it as a bug.

**Comment by andrewvaughan** [comment]
Can I at least ask why there's such an inconsistency? I have filetypes that leave double-spacing without issue, and then there's YAML which seems to always compress comments as compactly as possible and absolutely destroys readability. For example, these two files both pass Prettier linting:

No issue with multiple newlines, easily readable:
https://github.com/andrewvaughan/template-core/blob/main/.vale.ini

Significant and inconsistent compression of newlines, reduces readability, especially during indentation:
https://github.com/andrewvaughan/template-core/blob/3a83564b1c615151825a75491d64fc797b9e1fec/.mega-linter.yml

If anything, I would consider the inconsistency a bug if the intent of Prettier is to be extremely opinionated.
